src/client/dfuse/ops/readdir.c

```diff
@@ -130,50 +159,77 @@ dfuse_cb_readdir(fuse_req_t req, struct dfuse_inode_entry *inode,
 	} else {
 		if (offset != (off_t)&oh->doh_anchor)
 			D_GOTO(err, rc = EIO);
-		if (oh->doh_buf) {
-			fuse_reply_buf(req, oh->doh_buf, oh->doh_offset);
-			D_FREE(oh->doh_buf);
-			oh->doh_offset = 0;
+	}
+
+	/** if there was anything left in the old buffer */
+	if (offset && oh->doh_cur_off) {
+		/** if remaining does not fit in the fuse buf, return a block */
```

Comment:
I am not sure to totally grok this concept of block. Are we assuming 4k (i.e. one page) here?